es74x-cloud/src/main/scala/tech/beshu/ror/proxy/es/clients/HighLevelClientBasedClusterAdminClient.scala

```diff
@@ -0,0 +1,277 @@
+import org.elasticsearch.client.ClusterAdminClient
+import org.elasticsearch.cluster.{ClusterName, ClusterState}
+import org.elasticsearch.common.bytes.BytesReference
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.tasks.TaskId
+import org.elasticsearch.threadpool.ThreadPool
+
+class HighLevelClientBasedClusterAdminClient(esClient: RestHighLevelClientAdapter)
+                                            (implicit scheduler: Scheduler)
+  extends ClusterAdminClient {
+
+  override def health(request: ClusterHealthRequest): ActionFuture[ClusterHealthResponse] = ???
+
+  override def health(request: ClusterHealthRequest, listener: ActionListener[ClusterHealthResponse]): Unit = {
+    esClient
+      .health(request)
+      .runAsync(handleResultUsing(listener))
+  }
+
+  override def prepareHealth(indices: String*): ClusterHealthRequestBuilder = ???
+
+  override def state(request: ClusterStateRequest): ActionFuture[ClusterStateResponse] = ???
+
+  override def state(request: ClusterStateRequest, listener: ActionListener[ClusterStateResponse]): Unit = {
+    // todo: implement properly
+    listener.onResponse(new ClusterStateResponse(
+      ClusterName.DEFAULT,
+      ClusterState.EMPTY_STATE,
+      false
+    ))
+  }
+
+  override def prepareState(): ClusterStateRequestBuilder = ???
+
+  override def updateSettings(request: ClusterUpdateSettingsRequest): ActionFuture[ClusterUpdateSettingsResponse] = ???
+
+  override def updateSettings(request: ClusterUpdateSettingsRequest, listener: ActionListener[ClusterUpdateSettingsResponse]): Unit = ???
+
+  override def prepareUpdateSettings(): ClusterUpdateSettingsRequestBuilder = ???
+
+  override def prepareReloadSecureSettings(): NodesReloadSecureSettingsRequestBuilder = ???
+
+  override def reroute(request: ClusterRerouteRequest): ActionFuture[ClusterRerouteResponse] = ???
+
+  override def reroute(request: ClusterRerouteRequest, listener: ActionListener[ClusterRerouteResponse]): Unit = ???
+
+  override def prepareReroute(): ClusterRerouteRequestBuilder = ???
+
+  override def nodesInfo(request: NodesInfoRequest): ActionFuture[NodesInfoResponse] = ???
+
+  override def nodesInfo(request: NodesInfoRequest, listener: ActionListener[NodesInfoResponse]): Unit = ???
+
+  override def prepareNodesInfo(nodesIds: String*): NodesInfoRequestBuilder = ???
+
+  override def clusterStats(request: ClusterStatsRequest): ActionFuture[ClusterStatsResponse] = ???
+
+  override def clusterStats(request: ClusterStatsRequest, listener: ActionListener[ClusterStatsResponse]): Unit = ???
+
+  override def prepareClusterStats(): ClusterStatsRequestBuilder = ???
+
+  override def nodesStats(request: NodesStatsRequest): ActionFuture[NodesStatsResponse] = ???
+
+  override def nodesStats(request: NodesStatsRequest, listener: ActionListener[NodesStatsResponse]): Unit = ???
+
+  override def prepareNodesStats(nodesIds: String*): NodesStatsRequestBuilder = ???
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of ??? operator](https://app.codacy.com/app/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=4591272)